pkg/api/stars.go

```diff
@@ -42,25 +42,62 @@ func (hs *HTTPServer) GetStars(c *models.ReqContext) response.Response {
 //
 // Stars the given Dashboard for the actual user.
 //
+// Deprecated: true
+//
 // Responses:
 // 200: okResponse
 // 400: badRequestError
 // 401: unauthorisedError
 // 403: forbiddenError
 // 500: internalServerError
 func (hs *HTTPServer) StarDashboard(c *models.ReqContext) response.Response {
-	id, err := strconv.ParseInt(web.Params(c.Req)[":id"], 10, 64)
+	var id int64
+	var err error
+	id, err = strconv.ParseInt(web.Params(c.Req)[":id"], 10, 64)
 	if err != nil {
-		return response.Error(http.StatusBadRequest, "id is invalid", err)
+		return response.Error(http.StatusBadRequest, "Invalid dashboard ID", nil)
+	}
+	dash, rsp := hs.getDashboardHelper(c.Req.Context(), c.OrgID, id, "")
+	if rsp != nil {
+		return rsp
 	}
-	cmd := star.StarDashboardCommand{UserID: c.UserID, DashboardID: id}
 
-	if cmd.DashboardID <= 0 {
-		return response.Error(400, "Missing dashboard id", nil)
+	cmd := star.StarDashboardCommand{UserID: c.UserID, DashboardID: dash.Id, DashboardUID: dash.Uid}
```

Comment:
note that the dashboard UID is not stored in the `star` table (no such column); only the dashboard ID will be; so you may no need to call the `hs.getDashboardHelper()`

Reply:
You are right.